rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/Sw360AuthorizationController.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Siemens AG, 2018. Part of the SW360 Portal Project.
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.sw360.rest.authserver;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.sw360.datahandler.thrift.RestApiToken;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.rest.authserver.security.Sw360AuthenticationProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.stream.Collectors;
+
+@Controller
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class Sw360AuthorizationController {
+
+    @NonNull
+    private Sw360AuthenticationProvider sw360AuthenticationProvider;
+
+    // This service can only be called by localhost and should be protected
+    @RequestMapping(value = "/generateToken", method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.CREATED)
+    public @ResponseBody
+    RestApiToken generateToken(@RequestBody User sw360User) {
```

Comment:
This endpoint should have some way of authentication and authorization. Currently it is completly open and I think that "localhost is safe" is not the right approach.

Additionally it might be good to deactivate it by default.

Reply:
Yes you are right.
But how we can protect and fulfill authentication and authorization for this endpoint in a correct way?

The idea is to generate a token over sw360-portlet.
Since we don't want to use LDAP anymore we cannot user password as granted method.
In sw360-portlet we have `ActionRequest request` and we can extract the correct user by liferay in a correct way. How can we verify this information in the authorization server without having a dependency to liferay. Thanks.

Reply:
> How can we verify this information in the authorization server without having a dependency to liferay.

e.g. via a shared secret, or a public-private key.